libraries/AP_Scripting/applets/ScriptingSwitcher/Param_Controller.lua

```diff
@@ -0,0 +1,93 @@
+--[[
+   a script to select other parameters using an auxillary switch
+   from subdirectories in scripts directory labeled /1,/2, or /3
+--]]
+
+local SEL_CH = 302
+local PARAM_FILENAME = "params.param"
+
+--[[
+   check that directory exists
+--]]
+function check_subdir_exists(n)
+  return dirlist(get_scripts_dir() .. "/" .. n )
+end
+
+--[[
+   get the path to the scripts directory. This will be scripts/ on SITL
+   and APM/scripts on a ChibiOS board
+--]]
+function get_scripts_dir()
+   local dlist1 = dirlist("APM/scripts")
+   if dlist1 and #dlist1 > 0 then
+      return "APM/scripts"
+   end
+   -- otherwise assume scripts/
+   return "scripts"
+end
+
+--[[
+load parameters from a file PARAM_FILENAME from directory n
+--]]
+function param_load(n)
+   count = 0
+   failed = false
+   file_name = get_scripts_dir() .. "/" .. n .."/" .. PARAM_FILENAME
+  -- Open file
+  file = io.open(file_name)
+  if not file then
+     gcs:send_text(0,string.format("%s not present",file_name))
+     return
+  end
+   while true do
+      local line = file:read()
+      if not line then
+         break
+      end
+      -- trim trailing spaces
+      line = string.gsub(line, '^(.-)%s*$', '%1')
+      local _, _, parm, value = string.find(line, "^([%w_]+)%s*([%d]*.[%d]*)")
+      if parm then
+         if not param:set_and_save(parm,value) then
+            failed = true
+         else
+            count = count +1
+         end
+      end
+   end 
+   if not failed then
+      gcs:send_text(6,string.format("Loaded %u parameters",count)) 
+   else
+      gcs:send_text(6,string.format("Loaded %u parameters but some params did not exist to set",count)) 
```

Comment:
Could we also have a count of the number of params that failed? "... but %u params did not exist"

Reply:
I originally listed each param that failed...but it gets excessive sometimes in script conflicts...I think the warning is enough since you will probably have to do a reset to defaults and start over if you've conflicted in the scripts..

Reply:
Fair enough